File: backend-server/vikadata-service/vikadata-db-model/src/main/java/com/vikadata/entity/AutomationServiceEntity.java
package com.vikadata.entity;

import java.io.Serializable;
import java.time.LocalDateTime;

import com.baomidou.mybatisplus.annotation.FieldFill;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableLogic;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.experimental.Accessors;

/**
 * <p>
 * 自动化-服务表
 * </p>
 *
 * @author Mybatis Generator Tool
 */
@Data
@Builder(toBuilder = true)
@NoArgsConstructor
@AllArgsConstructor
@Accessors(chain = true)
@EqualsAndHashCode
@TableName("vika_automation_service")
public class AutomationServiceEntity implements Serializable {

    private static final long serialVersionUID = 1L;

    /**
     * 主键
     */
    @TableId(value = "id", type = IdType.ASSIGN_ID)
    private Long id;

    /**
     * 自定义服务ID
     */
    private String serviceId;

    /**
     * 服务可读唯一标识
     */
    private String slug;

    /**
     * 服务名
     */
    private String name;

    /**
     * 描述
     */
    private String description;

    /**
     * logo 地址
     */
    private String logo;

    /**
     * 基础 URL 地址
     */
    private String baseUrl;

    /**
     * 国际化语言包
     */
    private String i18n;

    /**
     * 删除标记(0:否,1:是)
     */
    @TableLogic
    private Boolean isDeleted;

    /**
     * 创建者
     */
    @TableField(fill = FieldFill.INSERT)
    private Long createdBy;

    /**
     * 最后修改者
     */
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private Long updatedBy;

    /**
     * 创建时间
     */
    private LocalDateTime createdAt;

    /**
     * 更新时间
     */
    private LocalDateTime updatedAt;

}

File: packages/room-server/src/automation/entities/automation.service.entity.ts
import { Column, Entity } from 'typeorm';
import { BaseEntity } from '../../shared/entities/base.entity';

@Entity(`automation_service`)
export class AutomationServiceEntity extends BaseEntity {
  @Column({
    name: 'service_id',
    nullable: false,
    length: 50,
  })
    serviceId: string;

  @Column({
    name: 'slug',
    nullable: false,
    length: 50,
  })
    slug: string;

  @Column({
    name: 'name',
    nullable: true,
    length: 255,
  })
    name: string;

  @Column({
    name: 'description',
    nullable: true,
    length: 255,
  })
    description: string;

  @Column({
    name: 'logo',
    nullable: true,
    length: 255,
  })
    logo: string;

  @Column({
    name: 'base_url',
    nullable: true,
  })
    baseUrl: string;

  @Column('json', {
    name: 'i18n',
    nullable: true,
  })
    i18n: object;
}
